packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.tsx

```diff
@@ -62,8 +63,18 @@ export const ButtonDownloadRecoveryKey = ({
     { supportURL }
   );
 
-  const fileContent = new Blob(
+  // 'keyWithContext' is currently disabled by default due to an issue with encoding recognition on Android.
+  // Non-latin text (e.g., Hebrew) may be displayed as jibberish due to incorrect encoding detection.
+  // In addition, localized strings contain important directionality markers that are hidden on Mac
+  // but visible on many other devices. On Apple devices, these directionality markers get copied with the key,
+  // and the key is rejected as invalid during password reset.
+
+  // Not tested - could possibly force recognition of UTF-8 encoding by adding BOM at the beginning of the blob
+  const BOM = new Uint8Array([0xef, 0xbb, 0xbf]);
```

Comment:
Does BOM mean anything in particular? Also if you want to test this in staging (if we're not planning to go to prod right now anyway), we could potentially enable this with a parameter instead of setting `keyWithContext` to `false`.

Reply:
BOM stands for "byte order mark". I've read that it can be used to identify the encoding of a text file, but I'm not sure if it would resolve the issue where some text is jibberish on Android.
https://learn.microsoft.com/en-us/globalization/encoding/byte-order-mark

Good call on enabling with a parameter so we can test if it makes a difference on stage.

Reply:
@vpomerleau Ah, TIL. Have you tried specifying `charset` in the type` or maybe there's another way to force utf-8? Like, `type: 'text/plain;charset=utf-8'`. Not sure if that'll work though.

You could maybe also try [TextEncoder](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder).